### ISSUE TITLE
[P1] Implement logit-space volatility estimator for binary markets

### ISSUE THREAD

**Issue by phrazzld** [phrazzld/atreides#6]
## Problem

Binary market prices are bounded [0, 1]. Standard volatility estimation (stddev of log-returns) breaks at the boundaries — as price approaches 0 or 1, returns become unbounded or undefined. The A-S strategy needs sigma (volatility) as input, and it must be estimated correctly for prediction markets.

The solution comes from "Toward Black-Scholes for Prediction Markets" (arxiv, Oct 2025): work in logit space.

## Context

- **Vision**: A-S market making requires volatility parameter sigma
- Logit transform: `x = ln(p / (1-p))` maps [0,1] to (-inf, inf)
- Belief volatility `sigma_b` = diffusion coefficient of logit price, not raw price
- The sigmoid naturally compresses spreads near boundaries — physically correct
- **Related**: #4 (reads snapshots from DataStore), #7 (scoring uses volatility)

## Acceptance Criteria

- [ ] Given a series of mid-prices [0.40, 0.42, 0.38, 0.45, 0.41], when computing rolling volatility, then result is in logit space (sigma_b)
- [ ] Given prices near boundaries (0.95, 0.96, 0.94), when computing volatility, then logit transform handles them correctly (no infinity, no NaN)
- [ ] Given prices at exact 0 or 1, when computing volatility, then they are clipped to [0.001, 0.999] before transform
- [ ] Given fewer than `window` samples, when computing volatility, then return NaN (not enough data)
- [ ] Given a DuckDB store with snapshot history, when `estimate_volatility(ticker, window)` is called, then sigma_b is computed from the stored mid-price time series

## Affected Files

- `src/atreides/analytics.py` — new file: `logit()`, `estimate_sigma()`, `estimate_sigma_ewma()`
- `tests/test_analytics.py` — new file: volatility estimation tests

## Verification

```bash
uv run pytest tests/test_analytics.py -v
uv run ruff check src/atreides/analytics.py
```

## Boundaries

- Do NOT implement jump-diffusion EM (Tier 3) — that's a later optimization
- Do NOT integrate with A-S strategy yet — just the estimator
- Do NOT add any trading logic

## Approach

Two tiers, implement both:

**Tier 1 — Rolling stddev (market scoring):**
```python
import math

def logit(p: float) -> float:
    p = max(0.001, min(0.999, p))
    return math.log(p / (1 - p))

def estimate_sigma(mid_prices: list[float], window: int = 60) -> float:
    logits = [logit(p) for p in mid_prices]
    returns = [logits[i] - logits[i-1] for i in range(1, len(logits))]
    if len(returns) < window:
        return float("nan")
    recent = returns[-window:]
    mean = sum(recent) / len(recent)
    var = sum((r - mean) ** 2 for r in recent) / (len(recent) - 1)
    return math.sqrt(var)
```

**Tier 2 — EWMA (live quoting, more reactive):**
```python
def estimate_sigma_ewma(mid_prices: list[float], span: int = 30) -> float:
    alpha = 2.0 / (span + 1)
    logits = [logit(p) for p in mid_prices]
    returns = [logits[i] - logits[i-1] for i in range(1, len(logits))]
    if not returns:
        return float("nan")
    ewma_var = returns[0] ** 2
    for r in returns[1:]:
        ewma_var = alpha * r**2 + (1 - alpha) * ewma_var
    return math.sqrt(ewma_var)
```

---
Created by `/groom`

**Comment by phrazzld** [comment]
Closing all GitHub Issues. Backlog management has moved to file-driven backlog.d/ items. Fresh grooming sessions will recreate what matters.
